### PR TITLE
Added mips architecture

### PR DESCRIPTION
**Pull request by SiD3W4y**

### CODE REVIEW

File: src/arch/mips.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+from asm.ast import *
+from asm.parse import *
+
+from functools import reduce
+
+class MipsArchitecture(Architecture):
+
+    @property
+    def name(self) -> str:
+        return "mips"
```
Comment (71) [inline src/arch/mips.py]: Can you please use single quotes everywhere instead of double ones? (for consistency)

File: src/arch/mips.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                    func = Function(name, [param("addr", TYPE_U32)])
+
+                    code = Literal(opcode << 26, TYPE_U32)
+                    truncated = Binary(OP_BITWISE_AND, 0x3ffffff, Binary(OP_SHL, Var("addr"), 2))
+                    
+                    func += Set(TYPE_U32, Binary(OP_BITWISE_OR, code, truncated))
+                    func += Increase(4)
+                    yield func
+                else:
+                    """
```
Comment (71) [inline src/arch/mips.py]: Once again for consistency, can you use a comment instead of a string here? Thanks!

**Review by 71 — CHANGES_REQUESTED** [review]
Looks good, but I'm a little picky so I'd like you to change two things before I try things out

**Comment by SiD3W4y** [comment]
Should be fixed

**Pull request merged**